src/controllers/userProfileController.js

```diff
@@ -594,10 +594,12 @@ const userProfileController = function (UserProfile) {
   const getUserById = function (req, res) {
     const userid = req.params.userId;
     if (cache.getCache(`user-${userid}`)) {
-      const getData = JSON.parse(cache.getCache(`user-${userid}`));
-      res.status(200).send(getData);
-      return;
-    }
+      const cachedData = JSON.parse(cache.getCache(`user-${userid}`));
+      if (cachedData.isUpToDate) {
```

Comment:
Hi, The property 'isUpToDate' does not exist in the user cached data. Consequently, this 'if' statement will always evaluate to false, which means 'getUserById' is no longer utilizing caching.